### PR TITLE
validations, model specs, and n+1 problems

### PR DESCRIPTION
**Pull request by tufoinnkuo10**
In this project the following was done
- Add the following validations:
   ** For the User model:
   -- Name must not be blank.
   -- PostsCounter must be an integer greater than or equal to zero.
  **For the Post model:
  -- Title must not be blank.
  -- Title must not exceed 250 characters.
  -- CommentsCounter must be an integer greater than or equal to zero.
  -- LikesCounter must be an integer greater than or equal to zero.
Add unit specs for all of your models' methods and validations.
Add flash messages in the create actions in all your controllers.
Make sure that the N+1 problem is solved when fetching all posts and their comments for a user

thanks for the review

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts.all
+    @posts = @user.posts
```
Comment (AshakaE) [inline app/controllers/posts_controller.rb]: - [x] Kindly resolve the n+1 query for posts by chaining the comments with it. see [reference](https://api.rubyonrails.org/v6.1.4.4/classes/ActiveRecord/QueryMethods.html#method-i-includes)

**Review by AshakaE — CHANGES_REQUESTED** [review]
### Hi @tufoinnkuo10  🙋‍♀️, 

Good job so far! 👌
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


#  Required Changes ♻️ 🚧
_Check the comments under the review._

###  Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (tufoinnkuo10) [reply app/controllers/posts_controller.rb]: resolved. thank you for your review.

**Review by Dipeshtwis — APPROVED** [review]
# Hi @tufoinnkuo10, 

### Status: Approved ✔️

:+1:  You have done well implementing the suggested changes. Well Done.

Your project is complete!  There are some optional changes I've suggested to you to make your project best, other than that nothing else to say.
Congratulations! 🎉 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- I would recommend adding the steps in the README file to run the project locally. [Hint: `bundle`, `rails db:setup`, `rails s`.


**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
-----

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Dipesh Kumar_**

------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**